### ISSUE TITLE
Feature Request : Custom Webhook

### ISSUE THREAD

**Issue by gregberger** [donorbox/donorbox-api#15]
Would be great to get the choice to bypass Zappier and get the same notifications on a custom webhook.

**Comment by campaignupgrade** [comment]
Ditto this. Would love a notification on new donation at the very least.

**Comment by claytonsmithnz** [comment]
Yes this would be amazing, we could integrate into Power Automate instead!

**Comment by adrianbj** [comment]
I actually can't believe this isn't already supported. I was just looking to have Donorbox notify my site when a payment is made and it seems like it's not currently possible which seems like a real limitation to this service. If I don't go through Zapier, is the only option to actually make regular calls to the API to get updates?

**Comment by campaignupgrade** [comment]
@adrianbj You could probably do something hacky like trigger an API call 200 seconds after the button is clicked. It would not be completely reliable though, and not completely real-time.

**Comment by adrianbj** [comment]
@campaignupgrade - thanks for the idea and certainly a possibility because I don't need it to be realtime, but it still seems like such an oversight for a service like this to not offer what is pretty basic functionality these days.

**Comment by campaignupgrade** [comment]
Agreed. What's irritating is that it's clear there's a webhook present but it only support Zapier. This really blocks our ability to integrate DonorBox with our CRM, email list manager, etc.

**Comment by adrianbj** [comment]
I wonder who has shares in who?  :)

Clearly Zapier is useful for non-tech folks, but for the rest of us, it's just an extra cost and layer of complexity. I went with DonorBox to save my dev time, not complicate things. End rant .......

**Comment by campaignupgrade** [comment]
I've found Zapier to be unstable. I've had Zaps fail without notification, error logs are essentially non-existent, and there doesn't seem to be a good way to monitor Zapier status externally. I can't put Zapier code in my repo so there's no effective way to unit test Zapier integration in a devops test-and-deploy workflow. 

Zapier is just not an appropriate tool for professional-level integration.

**Comment by adrianbj** [comment]
> Zapier is just not an appropriate tool for professional-level integration.

Completely agree - this really is an unacceptable situation.

**Comment by everton** [comment]
We are working to revamp our integrations and provide external URLs for the webhooks.

I expect to have it released during the next weeks. As soon as we release it, we will notify here.

**Comment by adrianbj** [comment]
Thanks so much @everton - this will be a huge help!

**Comment by harish86** [comment]
Hello everyone!

I am excited to inform you all that the custom webhooks fetaure has been released. More details on that can be found here  https://donorbox.zendesk.com/hc/en-us/articles/4733681068820-Custom-Webhooks.

We very much appreciate your patience on this. For any queries please feel free to contact our support support@donorbox.org.

**Comment by adrianbj** [comment]
@harish86 - this is great, but do we really need to pay $17 / month to be able to use it?

**Comment by claytonsmithnz** [comment]
Looks great, have linked it to Power Automate!
Question, how do we delete the webhooks? I cant seem to find this option.

**Comment by zarqiqbal** [comment]
@claytonsmithnz they can be deactivated for now through a toggle. For all practical purposes that will work but we will consider the ability to remove the webhook altogether later.

**Comment by HugoGresse** [comment]
Following up on the pricing of the Zapier & API. 
Not providing an API seems absurd to me. The goal of the API is to provide an alternative for the website. 
Donorbox is already taking a cut on each donation, it's fair, but it should also provide bare minimal feature while taking the cut, and providing the API (not zapier) should be free of charge. 

Failing to do that, we'll need to setup webscrapping to provide an alternative API :/

**Comment by joshuagraber** [comment]
> Failing to do that, we'll need to setup webscrapping to provide an alternative API :/

Good luck scraping. I'm trying to do a CORS proxy of this thing so that it can work within a web application, and even my DigitalOcean IP is blocked by DonorBox's aggressive Cloudfront 🙃

**Comment by HugoGresse** [comment]
Well, I'm used to cloudfront and bypassing it through my personnal network so it's not an issue.
But in this case we've finally payed for the Webhook integration. 

I think the tech stack on Donorbox is lacking a lot, and need a lot of improvement, if someone at Donorbox want to chat with me, let me know.
